src/tests/homework.spec.js

```diff
@@ -0,0 +1,68 @@
+import { test } from "@playwright/test";
+import { EnrollmentPage } from "./homework.enrollment.page";
+import { password, username, userFullName } from "../fixtures/fixtures";
+
+test.describe("Registration form", async () => {
+
+    test.beforeEach(async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+
+        await enrollmentPage.navigate();
+        await enrollmentPage.assertPageHeading("Registrace");
+    });
+
+    test("Registration form should be visible, enabled and editable", {
+        annotation: {
+            type: "test description",
+            description: "Test který přejde na formulář registrace a zkontroluje, že se formulář správně zobrazil"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
+
+        await enrollmentPage.assertFormElements();
+    });
+
+    test("New user should register", {
+        annotation: {
+            type: "test description",
+            description: "Test který provede validní registraci uživatele"
+        }
+    }, async ({ page }) => {
+        const enrollmentPage = new EnrollmentPage(page);
```

Comment:
v každém objektu vytváříš novou instanci objektu, ale máš to zároveň i v beforeEach, což nadbytečné. Jde to vyřešit následovně:
pod test.describe... přidej:
let enrolmentPage;

tim se ti definuje ta stranka -> let je variable a pak diky beforeEach se ti pro kazdy test udela novej context page -> nemusis se bat, ze by se objekty v testech nejak prepisovaly. Pote pak muzes odstranit   const enrollmentPage = new EnrollmentPage(page); z kazdeho testu

Reply:
Upraveno, díky!